sdk/iot/provisioning/tests/cmocka/test_az_iot_provisioning_client_parser.c

```diff
@@ -466,6 +469,55 @@ static void test_az_iot_provisioning_client_operation_complete_translate_succeed
   assert_true(az_iot_provisioning_client_operation_complete(AZ_IOT_PROVISIONING_STATUS_DISABLED));
 }
 
+static const az_span _log_received_topic = AZ_SPAN_LITERAL_FROM_STR("$dps/registrations/res/202");
+static const az_span _log_received_payload = AZ_SPAN_LITERAL_FROM_STR("LOG_PAYLOAD");
+
+static int _log_invoked_topic = 0;
+static int _log_invoked_payload = 0;
+static void _log_listener(az_log_classification classification, az_span message)
+{
+  switch (classification)
+  {
+    case AZ_LOG_MQTT_RECEIVED_TOPIC:
+      assert_memory_equal(
```

Comment:
Is it feasible to use assert_string_equal here so that if the assert fails we can see the expected vs actual?

If `assert_memory_equal` also show that comparison, then ignore this.

Reply:
Unfortunately az_span is not a string and so doesn't guarantee the null terminator.
UTs are used as programming patterns so I don't want to give the wrong impression (in this particular case, but not others - see the sas token logger- that would not be true).